### ISSUE TITLE
Installation steps with AUR

### ISSUE THREAD

**Issue by raven2cz** [jaelpark/chamferwm#20]
Hello, I'm trying to install chamfer from aur, but the installation fails on shaderc-git which throws error without message, just a failure occured in prepare(), nothing more. 
Can you help me with installation? I can try to compile from meson, but how about deps for running after that?
Thanks for help.

**Comment by jaelpark** [comment]
Hi, thanks for the report. I just tried rebuilding shaderc-git, and it doesn't seem to build for me either. However, I tested chamfer with shaderc from the official repo, and looks like it might work again. I think you can also try that one, see how it goes. I will remove the shaderc-git recommendation from the aur page.

**Comment by raven2cz** [comment]
Thanks for response. So, please update aur to drop shaderc-git dep. I will try it after that.

**Comment by jaelpark** [comment]
Done, forgot it wasn't done already.

**Comment by raven2cz** [comment]
I already compile and install it from AUR. Thanks. But it cannot start. Missing some deps? See log.

```
❯ cat log.txt
No pulsectl module.
[chamferwm 2020-07-27 11:33:57] Loading config /usr/share/chamfer/config/config.py...
[chamferwm 2020-07-27 11:33:57] Screen size: 3840x2160 (DPI: 96.000000x0.000000)
[chamferwm 2020-07-27 11:33:57] Root id: 6c8
[chamferwm 2020-07-27 11:33:57] Backend initialized.
[chamferwm 2020-07-27 11:33:57] XComposite 0.4
[chamferwm 2020-07-27 11:33:57] overlay xid: 128
[chamferwm 2020-07-27 11:33:57] XFixes 5.0
[chamferwm 2020-07-27 11:33:57] Damage 1.1
[chamferwm 2020-07-27 11:33:57] SHM 1.2
[chamferwm 2020-07-27 11:33:57] DRI3 1.2
[chamferwm 2020-07-27 11:33:57] Enumerating required layers
[chamferwm 2020-07-27 11:33:57] Enumerating required extensions
  VK_EXT_debug_report
```

```
❯ cat .xinitrc
numlockx &
xrdb -merge ~/.Xresources
export QT_STYLE_OVERRIDE=kvantum
export QT_QPA_PLATFORMTHEME="qt5ct"
export DESKTOP_SESSION=plasma
export XDG_CURRENT_DESKTOP=KDE
#herbstluftwm
#exec i3
#exec qtile
#exec gnome-session
#exec startplasma-x11
exec chamfer --config=/usr/share/chamfer/config/config.py --shader-path=/usr/share/chamfer/shaders/
```

**Comment by jaelpark** [comment]
Is that the complete log? Can you try run with all buffers logged?
`exec stdbuf -oL -eL chamfer --config=/usr/share/chamfer/config/config.py --shader-path=/usr/share/chamfer/shaders/ > /tmp/log 2>&1`

Additionally, you can install `vulkan-validation-layers` and run the with validation layers enabled, see if there are any strange complaints:
`exec stdbuf -oL -eL chamfer -l --config=/usr/share/chamfer/config/config.py --shader-path=/usr/share/chamfer/shaders/ > /tmp/log 2>&1`

If that doesn't work, see if the compositor is able to start in a debug window. Do this inside any other window manager:
`chamfer -d --config=/usr/share/chamfer/config/config.py`
There you should be able to create "windows" by pressing space.

**Comment by jaelpark** [comment]
By the looks of it some of the vulkan extensions can not be loaded. What's your GPU? Probably the loaders are missing somehow. Are you able to run other vulkan applications?

**Comment by raven2cz** [comment]
Yes, you have right. I have to install "vulkan-radeon" for my radeon 5700.
 your code
`chamfer -d --config=/usr/share/chamfer/config/config.py`
works it shows new window and after pressing spacebar, the window with shadow is placed correctly.

Log for your first test line is here:
[chamfer.log](https://github.com/jaelpark/chamferwm/files/4981624/chamfer.log)

I already started chamfer. The mouse pointer and whole white screen is shown. But it is all, there is no work keybinding or something else. I unable to work with WM yet. 
What has to be shown after first launch? Or what has to be configure, I checked the config.py, it seems that there is a lot of things which can be call... :(
I tried keybinding for termite for example enter+mod but nothing happen....

**Comment by jaelpark** [comment]
Glad to hear it's running! The screen is white because a wallpaper or background hasn't been set (feh or any other tool should work fine).

By default Alt is the mod key and Alt+Enter is supposed to bring up the terminal. In the config you can set your preferred terminal by changing just [this line](https://github.com/jaelpark/chamferwm/blob/710eecbf3a6960650ffa608f9ad5fdbe8b170ef4/config/config.py#L571). Try also other combintions, such as Win+Enter, although not sure why they would be mixed up. Are you able to launch a web browser for example? By default Win+1 launches firefox, [but that you can change too](https://github.com/jaelpark/chamferwm/blob/710eecbf3a6960650ffa608f9ad5fdbe8b170ef4/config/config.py#L574). Make sure that python psutil launches processes correctly🤔. I'm hoping it's not a compositor issue. You can also try chamfer without the compositor with `exec chamfer -n --config=/usr/share/chamfer/config/config.py` and check whether applications show up in that environment.

To exit the WM, Alt+Shift+E is bound, you can test if at least that works.

**Comment by raven2cz** [comment]
no, it seems that the problem will be in python keybinding. I have python 3.8 as default now. psutil is installed. I'm looking to imports in config.py. It seems that all libs I have installed instead one:  from enum import Enum,auto  => enum.
I cannot install it, because error  AttributeError: module 'enum' has no attribute 'IntFlag'
I don't know if this will be the error. But the script was called in the test too, right?

`chamfer -d --config=/usr/share/chamfer/config/config.py`

and there worked the 'spacebar' for show window. If there will be problem with imports, the chamfer is not started, or?

In any case, which libraries has to installed with pip? Maybe it will be my problem.

**Comment by jaelpark** [comment]
Actually, in the test the window key combination is still hard coded, so it wasn't affected by this Enum problem. Good that you have spotted it! I believe this is indeed what prevents the scripts from running.

I'm not completely sure how to solve it, but it might be that you probably have some python package installed that is conflicting with the defaults. I found something related to this, might be helpful in your case:
https://stackoverflow.com/questions/58890684/attributeerror-module-enum-has-no-attribute-intflag-fastai-install
https://stackoverflow.com/questions/47878060/why-is-the-re-module-trying-to-import-enum-intflag

**Comment by raven2cz** [comment]
Sorry, but I haven't problem with enum34. Enum.py (version 3.4) is installed with python38 as part in bin folder enum.py. It seems that cannot be installed separatelly. I don't find the problem. Maybe there will be good to create some checking script that the config.py is correctly configured and it is possible to use it with operating system.
For example, qtile has checking by this way:
`python2 -m py_compile ~/.config/qtile/config.py`
I'm not sure if the problem is enum lib. But I'm blind now and cannot see any error.
